### ISSUE TITLE
Responding with an error to an XMPP stanza with partial addressing fails.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2614]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#2614|
|Type|defect|
|Created|2007-05-01 12:10:50Z|

<details><summary>Searchable metadata</summary>

```
trac-id__2614 2614
type__defect defect
reporter__ralphm ralphm
priority__normal normal
milestone__Words_0_6 Words-0.6
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1178021450000000 1178021450000000
changetime__1178198290000000 1178198290000000
version__None None
owner__ 
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm commented|
|-|-|

This is being addressed in [#2613](https://github.com/twisted/twisted/issues/2613).

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm set status to `closed`|
|-|-|

(In [20178]) Provide function for creating XMPP response stanzas.

Author: ralphm
Reviewer: therve
Fixes [#2613](https://github.com/twisted/twisted/issues/2613), [#2614](https://github.com/twisted/twisted/issues/2614).

This adds `twisted.words.protocols.jabber.xmlstream.toResponse` for creating a
response stanza from a received stanza with reversed addressing. It also
ensures stanzas with partial addressing will have a proper response, which was
a problem for `twisted.words.protocols.jabber.error.StanzaError.toResponse`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
